src/main/java/com/google/maps/PendingResultBase.java

```diff
@@ -89,32 +91,32 @@ public final void cancel() {
   protected abstract void validateRequest();
 
   protected A param(String key, String val) {
-    params.put(key, val);
+    if(params.get(key) == null) {
+      params.put(key, new ArrayList<String>());
```

Comment:
This looks naïvely fine but I wonder if anyone is using this class to overwrite previous values being set via `.param()`.

Would it be worth having an explicit 'add' flag somehow? I know you already have the three `param` methods, so a matrix of those would be awkward. Let me know.

Reply:
In fact, yes I was being naive. There was logic in optimise waypoints that implicitly relied on that underlying hashmap behavior of the parameters. 

New implementation now enforces singleton parameters (as implicitly required by non Static Maps API surfaces), and just uses the multiple parameters for Static Maps `path`s and `markers`.

Now the tests pass...